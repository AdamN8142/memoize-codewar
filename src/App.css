@import url('https://fonts.googleapis.com/css?family=Cinzel+Decorative');


* {
  margin:0;
  padding:0;
  box-sizing: border-box;
  font-weight: bold;
}

.app {
  background-image:url("https://wallpapercave.com/wp/gfwPusm.jpg");
  height: 100vh;
}