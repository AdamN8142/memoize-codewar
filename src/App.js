import React, { Component } from 'react';
import FilterControls from './FilterControls'
import Header from './Header'


class App extends Component {
  constructor() {
    super();
    this.state = {
      data: [],
      completed: []
    }
  }

  toggleComplete = (id) => {
    let updatedData = this.state.data.map((card) => {
      if (card.id === id) {
        card.addedToStorage = !card.addedToStorage;
      }
        return card
    })
    let updatedCompletedArr = updatedData.filter( card => card.addedToStorage).map(card=> card.id)
    this.setState({
      data: updatedData, 
      completed: updatedCompletedArr
    }, this.setLocalStorage)
  }

    setLocalStorage = () => {
      localStorage.setItem('completed', JSON.stringify(this.state.completed))
  }

  getLocalStorage = () => {
    if(localStorage.getItem('completed')){
      let storage = JSON.parse(localStorage.getItem('completed'))
      let updatedData =this.state.data.map((card) => {
        if(storage.includes(card.id)) {
          card.addedToStorage = true;
        }
          return card
      })
      this.setState({
        data: updatedData,
        completed:storage
      })
    } 
  }
    
  componentDidMount = () => {
    fetch("http://memoize-datasets.herokuapp.com/api/v1/ANdata")
      .then(response => response.json())
      .then(data => {
        this.setState({
          data: data.ANdata
        }, () => this.getLocalStorage())
      })
      .catch(error => {
        throw new Error(error);
      });
  }

  render() {
        return (
          <div className="app">
            <Header />
            <FilterControls 
              data={this.state.data} 
              toggleComplete={this.toggleComplete} 
              completed={this.state.completed}/>
          </div>
        )
    }
}


export default App;
