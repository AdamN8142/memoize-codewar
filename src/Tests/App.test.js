import React from 'react';
// import ReactDOM from 'react-dom';
import App from '../App';
import { shallow } from "enzyme";

describe("App", () => {
  let wrapper;

  beforeEach(()=> {
    wrapper = shallow(
      <App />
    )
  })

  it("should have the proper default state", () => {
    expect(wrapper.state()).toEqual({data: [], completed:[]})
  })

  it("should match the snapshot with all data passed in", () => {
    expect(wrapper).toMatchSnapshot();
  })
});