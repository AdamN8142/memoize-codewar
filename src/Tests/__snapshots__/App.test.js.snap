// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot with all data passed in 1`] = `
<div
  className="app"
>
  <Header />
  <FilterControls
    completed={Array []}
    data={Array []}
    toggleComplete={[Function]}
  />
</div>
`;
